ons-spark/spark-analysis/big-data-workflow.md

```diff
@@ -0,0 +1,118 @@
+# **Big data workflow**
+Working with big data efficiently presents a major challenge to analysts. In particular, working with large datasets can use large amounts of computational resource, which can be very expensive. Inefficient code or working practices can unnecessarily increase the costs of running your analysis, as well as the time it takes you to run your code. However, there are good practices that can be employed to help minimise costs without sacrificing the quality of your analysis. We have produced this page as a general guide to best practice in working with big data, including tips for general workflow and managing pipeline development, working with samples and subsets of your data, and optimising analysis with Spark.
+
+
+## **General workflow and project planning**
+
+### 1. Plan your analysis
+- A simplified version of the process of developing a new pipeline might look something like this:
+
+
+```{figure} ../images/big_data_pipeline_stages.png
+---
+width: 100%
+name: BigDataPipelineStages
+alt: Diagram showing potentials steps in the process of developing a new big data pipeline.
+---
+Fig 1. Big data pipeline development stages
+```
+
+- Splitting your development into clearly defined stages like this will help with applying some of the tips we have below for keeping your resource use to a minimum. For example, working with small samples or subsets of your data in the EDA phase.
```

Comment:
I initially wrote a comment here to suggest that they should probably be using samples in stages 2/3 too, but then noticed this is mentioned later. Still, I think it's a good idea to mention here too.

Reply:
Good idea - edited to also mention stages 2 and 3 here.